Source/Scene/PostProcessorScene.js

```diff
@@ -20,23 +20,247 @@ define([
+
+            'float red = 0.0; \n' +
+            'float green = 0.0; \n' +
+            'float blue = 0.0; \n' +
+            'if(bSpace) \n' +
+            '{ \n' +
+                'red = IsinEarth(texcoord  + direction * distortion.r, SceneSize) * texture2D(tex, texcoord + direction * distortion.r).r; \n' +
+                'green = IsinEarth(texcoord + direction * distortion.g, SceneSize) *  texture2D(tex, texcoord + direction * distortion.g).g; \n' +
+                'blue = IsinEarth(texcoord  + direction * distortion.b, SceneSize) * texture2D(tex, texcoord + direction * distortion.b).b; \n' +
+            '} \n' +
+            'else \n' +
+            '{ \n' +
+                'red = texture2D(tex, texcoord + direction * distortion.r).r; \n' +
+                'green = texture2D(tex, texcoord + direction * distortion.g).g; \n' +
+                'blue = texture2D(tex, texcoord + direction * distortion.b).b; \n' +
+            '} \n' +
+
+
+
+            'return vec4(clamp(red, 0.0, 1.0), clamp(green, 0.0, 1.0), clamp(blue, 0.0, 1.0), 0.0); \n' +
+
+            '}   \n' +
+
+
+
+            'void main(void) \n' +
+            '{ \n' +
+            '    vec3 rgb = texture2D(u_colorTexture, v_textureCoordinates).rgb; \n' +
+
+
+            'bool bSpace = true; \n' +
+
+            //whether it is in space or not
+            'if(length(czm_viewerPositionWC.xyz) < 6500000.0) \n' +
+            '   bSpace = false; \n' +
+
+            //Sun position
+            'vec4 SunPos; \n' +
+            'if (czm_morphTime == 1.0) \n' +
+            '{                         \n' +
+            '    SunPos = vec4(czm_sunPositionWC, 1.0);   \n' +
+            '}                                              \n' +
+            'else \n' +
+            '{ \n' +
+            '    SunPos = vec4(czm_sunPositionColumbusView.zxy, 1.0); \n' +
+            '} \n' +
+
+            'vec4 SunPositionEC = czm_view * SunPos; \n' +
+            'vec4 SunPositionWC = czm_eyeToWindowCoordinates(SunPositionEC); \n' +
+
+
+            'SunPos = czm_viewportOrthographic * vec4(SunPositionWC.xy, -SunPositionWC.z, 1.0); \n' +
+            'vec2 texcoord = -v_textureCoordinates + vec2(1.0);   \n' +
+
+            'vec2 texelSize = 1.0 / czm_viewport.zw;    \n' +
+            'vec3 distortion = vec3(-texelSize.x * u_Distortion, 0.0, texelSize.x * u_Distortion);    \n' +
+
+            // ghost vector to image centre:
+            'vec2 ghostVec = (vec2(0.5) - texcoord) * u_GhostDispersal;  \n' +
+            'vec3 direction = normalize(vec3(ghostVec, 0.0));  \n' +
+
+            // sample ghosts:  
+           ' vec4 result = vec4(0.0);  \n' +
+           ' vec4 ghost = vec4(0.0);  \n' +
+           ' for (int i = 0; i < 4; ++i) {  \n' +
+           '     vec2 offset = fract(texcoord + ghostVec * float(i));  \n' +
+
+                //Only bright spots from the centre of the source image
+           //'     float weight = length(vec2(0.5) - offset) / length(vec2(0.5));  \n' +
```

Comment:
Delete commented out code.

If you really need it, it could stay for now, but once this is merged into `master`, we do not leave in commented out code.